# Lab 1: Exploratory Data Analysis

by Nino Castellano, Will Lahners, Eddie Powers

### 1. Business Understanding

*** eddies part***

============================

[Dataset: Automobile Sales | Visual EDA](https://www.kaggle.com/code/ddosad/automobile-sales-visual-eda)

### 2. Data Understanding

##### 2.1 Data Description

In [2]:
#Required Libraries
import numpy as np
import pandas as pd

import seaborn as sns
import matplotlib.pyplot as plt
import plotly.graph_objects as go
import plotly.express as px

from warnings import filterwarnings
filterwarnings('ignore')

In [14]:
#Loading the file
df = pd.read_csv("https://raw.githubusercontent.com/ncastellano2320/Machine-Learning-in-Python/86d705d3789e7cbfeb3ecabcbd34709c116b456c/Lab1_ExploringTableData/Auto%20Sales%20data.csv?token=GHSAT0AAAAAACNGPIZJTKC4KS2NBQD6INVAZN5MMQQ", parse_dates=['ORDERDATE'], dayfirst=True)

#Replace the deal size to represent small, medium, and large with "1", "2" or "3" respectively. 
df.DEALSIZE.replace(to_replace = ['Small', 'Medium', 'Large'],
                 value = range(1,4), inplace = True)

#Drop unnecessary data
df = df.drop(['PHONE', 'CUSTOMERNAME', 'CONTACTFIRSTNAME', 'CONTACTLASTNAME', 'ADDRESSLINE1', 'POSTALCODE', 'STATUS'], axis=1)
df.head()

ORDERNUMBER  QUANTITYORDERED  PRICEEACH  ORDERLINENUMBER    SALES  \
0        10107               30      95.70                2  2871.00   
1        10121               34      81.35                5  2765.90   
2        10134               41      94.74                2  3884.34   
3        10145               45      83.26                6  3746.70   
4        10168               36      96.66                1  3479.76   

   ORDERDATE  DAYS_SINCE_LASTORDER  PRODUCTLINE  MSRP PRODUCTCODE        CITY  \
0 2018-02-24                   828  Motorcycles    95    S10_1678         NYC   
1 2018-05-07                   757  Motorcycles    95    S10_1678       Reims   
2 2018-07-01                   703  Motorcycles    95    S10_1678       Paris   
3 2018-08-25                   649  Motorcycles    95    S10_1678    Pasadena   
4 2018-10-28                   586  Motorcycles    95    S10_1678  Burlingame   

  COUNTRY  DEALSIZE  
0     USA         1  
1  France         1  
2  France         2  
3     USA         2  
4     USA         2

Not only did we drop some unnecessary data, but we also changed the data type of the "DEALSIZE" variable, as it makes more sense to be processed as integers to represent small, medium, and large with "1", "2" or "3", respectively. 

In [15]:
#Find the data type
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2747 entries, 0 to 2746
Data columns (total 13 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   ORDERNUMBER           2747 non-null   int64         
 1   QUANTITYORDERED       2747 non-null   int64         
 2   PRICEEACH             2747 non-null   float64       
 3   ORDERLINENUMBER       2747 non-null   int64         
 4   SALES                 2747 non-null   float64       
 5   ORDERDATE             2747 non-null   datetime64[ns]
 6   DAYS_SINCE_LASTORDER  2747 non-null   int64         
 7   PRODUCTLINE           2747 non-null   object        
 8   MSRP                  2747 non-null   int64         
 9   PRODUCTCODE           2747 non-null   object        
 10  CITY                  2747 non-null   object        
 11  COUNTRY               2747 non-null   object        
 12  DEALSIZE              2747 non-null   int64         
dtypes: datetime64[ns](

Based on the above output of the dataframe, there are no missing values within our dataset as every column's non-null count adds up to the total number of rows in the data, 2747. 

Because there are no missing rows, it would be unnecessary to do any imputation on the data, as it is a complete dataset. 